### ISSUE TITLE
Editor Question : Is there a way to automatically save projects with multiple 'backup' copies?

### ISSUE THREAD

**Issue by randibit** [godotengine/godot#29393]
I'm relatively new to coding / Godot.  I am taking a Godot course and make mistakes not always realizing which one breaks the Game.

I do backup the project folder often, but with the many changes even in one lesson, it's never enough.
Projects are overwritten when hitting F5 (running project).  So this needs to be a save locally (fast) before the next change.

If there is a feature I don't know about that can do this, I would appreciate instruction.

If not, could one be added?

Thanks

Example draft of what backups might look like from Godot:
![image](https://user-images.githubusercontent.com/50721970/58757477-de05fe00-84c1-11e9-8556-9757f345884d.png)

**Comment by TheDuriel** [comment]
You are making this issue on a website dedicated to keeping incremental backups of code projects.

https://rogerdudler.github.io/git-guide/

**Comment by randibit** [comment]
Haha, I asked about Godot features not Github =D

**Comment by TheDuriel** [comment]
There is no obligation to use GIT with a cloud service. It is in fact heavily used for offline versioning. Comitting your changes is done with a single press of a button in nearly every visual interface. Github provides a newby friendly one for free, https://desktop.github.com/

**Comment by groud** [comment]
Yes, this is already possible using any version control system (VCS) out there. You can use any of them such as git, svn, perforce or mercurial.

VCS integration to the editor is even planed as a gsoc project.

If you have more questions please ask on the other community channels, as Github is dedicated to engine bugs and features proposal.

**Comment by randibit** [comment]
I don't want to use Git, I want it in Godot, that was the point.

I'll write my own tool to do it.

**Comment by groud** [comment]
> I don't want to use Git, I want it in Godot, that was the point. I'll write my own tool to do it.

Git is a "version control system", it is exactly meant to achieve what you are looking for. 
Godot won't include a dedicated VCS, it is out of it's scope.

However, as I just mentioned, one of the gsoc project aims at integrating git (or any other VCS) into the editor, so you can use a simple interface to manage your versions.

**Comment by randibit** [comment]
I look forward to that feature, thanks

**Comment by randibit** [comment]
I found a workaround for this.

**Comment by randibit** [comment]
Just fyi.  I worked in support/triage situations for a long time.  I would have answered the question like this.
No we don't have the feature yet.  We are working on it.
Straightforward, to the point and no embellishment.
Have a good one.

**Comment by randibit** [comment]
Looks like the trolls used confused emoji.  No one ever answered the question I asked.  Not even close.  I ask Unity for help and no sarcasm, but an answer to the question I ask.  UE4, same thing. This is how it should be.  Treat the customer right so you start having some.

**Comment by groud** [comment]
Sorry, the Godot organization is not a company, and you are not a customer.
 
Also, as I said before, this is not the place to ask questions or ask for help anyway, please use the other community channels for this purpose.

**Comment by randibit** [comment]
Ironic, as I asked in the form of a question after being trolled for reporting bugs.  The arrogance is off the scale.

**Comment by akien-mga** [comment]
@randibit I have to remind you that we have a Code of Conduct: https://godotengine.org/code-of-conduct

Please stay constructive.

**Comment by randibit** [comment]
> Code of Conduct: https://godotengine.org/code-of-conduct

Please pass it on to the others.  TY.

**Comment by randibit** [comment]
> There is no obligation to use GIT with a cloud service. It is in fact heavily used for offline versioning. Comitting your changes is done with a single press of a button in nearly every visual interface. Github provides a newby friendly one for free, https://desktop.github.com/
> 
> I don't see any reason why Godot should clumsily backup folders, when git is basically the industry standard for software project versioning already. And Godot is in fact _more git friendly than basically any other engine out there_.

I have to remind you that we have a Code of Conduct: https://godotengine.org/code-of-conduct

Please stay constructive.

**Comment by randibit** [comment]
> You are making this issue on a website dedicated to keeping incremental backups of code projects.
> 
> https://rogerdudler.github.io/git-guide/

Not constructive,

**Comment by randibit** [comment]
Noticed what the trolling was about, you copied the suggestion, including the fix into 3.2.2 giving credit do one of the contributing devs.  Not the first time I had people steal the credit for my work.

**Comment by akien-mga** [comment]
I don't know what you're talking about since we did not implement any automatic backup project copy feature in 3.2.2. There's also no "work" to credit to you even if we did, making a suggestion is not implementing a feature.

Anyway, as mentioned in https://github.com/godotengine/godot/issues/29393#issuecomment-498019121, if you want to make backups in a convenient way, the most appropriate solution is to use Version Control Systems like Git, SVN, Mercurial, Perforce, Plastic, etc. These systems have been designed *exactly* for this, and make a hard copy of project files automatically would be a waste of disk space, I/O and CPU.

So once again, we're not interested in implementing your suggestion. Feel free to do it yourself for your own use if you really need it. I'll lock this thread as everything as been said, and you posting a random comment every few months doesn't really help anyone.
